plugins/janus_audiobridge.c

```diff
@@ -3905,6 +4008,9 @@ static void *janus_audiobridge_handler(void *data) {
 			if(participant->opus_pt < 0) {
 				/* TODO Handle this case */
 				JANUS_LOG(LOG_ERR, "Offer doesn't contain Opus..?\n");
+			} else {
+				participant->fec = janus_sdp_attribute_fec_enable(offer, participant->opus_pt);
```

Comment:
There are a couple of issues here:

1. `participant->fec` here is set to `TRUE` only when we're working on the answer, which is *after* we handled the `join` most of the times; this means that when we set up the encoder and decoder, `participant->fec` is still `FALSE`. A check on whether or not fec is enabled should be done in the `join` handler code instead, and then used for building the SDP answer as well later on.
2. I personally don't like the idea of a new ad-hoc method in the SDP utils for this FEC thing: the semantics used here are very specific to the Opus codec, and not general to other codecs as well. The `janus_sdp_attribute_*` stuff should be used here instead, without touching the utils. I'll add a comment later showing you how you can detect if FEC is being negotiated by the other party, and how to add the `;useinbandfec=1` part to your answer.

EDIT: actually, looking at the code below, we're already adding the `useinbandfec` part, so what's missing is just the detection of whether or not the user wants to do FEC. This is actually quite easy, and doesn't require any change to SDP utils at all: a simple `strstr(msg->jsep_sdp, "useinbandfec=1")` in the `join` code should do the trick.

Reply:
1. I doesn't understand, I'm looking at the SDP offer (janus_sdp *offer ....), then we will prepare the answer according from the fec information that we get from offer. You get opus pt from offer too
2. I can use if you want a strstr() and remove all utils change

Reply:
Yes, what I meant is that the new utils change to detect FEC information is overkill (a strstr is probably enough), and actually wouldn't belong to the SDP utils as too codec specific.